manifests/source.pp

```diff
@@ -32,6 +32,10 @@
     if ! $location {
       fail('cannot create a source entry without specifying a location')
     }
+    $method = split($location, '[:\/]+')[0]
```

Comment:
This is an odd regex because it'll also match `:/:/` but I guess it works. You could also compare it to the `Stdlib::Httpsurl` type to verify it starts with `https://`

Reply:
I looked at coding this approach. However because location is just a string, the comparison would not work. location can pretty much be anything. 
IE the following will fail
```
if type(location) == Stdlib::HTTPSUrl {
      ensure_packages('apt-transport-https')
    }
```
Even changing the definition of location at the top will mean it will always be reported as a string.

Reply:
Re $_transport_https_releases = [ 'wheezy', 'jessie', 'stretch', 'trusty', 'xenial' ]
Shakes head, i dislike lists like this as they never get updated. However i missed the point that this is a finite list. With newer oses not needing the package. 

Great catch @ekohl 👍

Reply:
Wouldn't `if $location =~ Stdlib::Httpsurl` work?
The following does call fail:
```puppet
$x = 'https://fsd'
if $x =~ Stdlib::Httpsurl {
  fail('http only')
}
```

Reply:
It does work, apologies. I did find another bug though.  This will not hit the fail
```
$x = 'HTTPS://fsd'
if $x =~ Stdlib::Httpsurl {
  fail('http only')
}
```